servers/rendering/shader_compiler.cpp

```diff
@@ -381,7 +381,12 @@ void ShaderCompiler::_dump_function_deps(const SL::ShaderNode *p_node, const Str
 	}
 }
 
-static String _get_global_shader_uniform_from_type_and_index(const String &p_buffer, const String &p_index, ShaderLanguage::DataType p_type) {
+static String _get_global_shader_uniform_from_type_and_index(const String &p_buffer, const String &p_index, ShaderLanguage::DataType p_type, bool p_use_uint_as_index = false) {
+	String index_suffix;
+	if (p_use_uint_as_index) {
+		index_suffix = "u";
+	}
```

Comment:
I see that this function is only called in 4 places. Two are using global variables and 2 are using instance variables. Is there any reason that instance variables shouldn't have "u" appended to the 1? 

If not, the "u" could just be added to the string which would avoid doing these string operations at run time.

Reply:
No

Reply:
I think we can most likely just always add the "u"

Reply:
Agreed!